src/hip_ra_x/hip_ra_x.py

```diff
@@ -905,13 +919,107 @@ def render_scientific(p: floatParameter | OutputParameter) -> str:
             self.logger.critical(msg)
             raise
 
-        # copy the output file to the screen
-        with open(outputfile, encoding='UTF-8') as f:
-            content = f.readlines()  # store all output in one long list
+        if self.html_output_file.Provided:
+            # write the outputs to the output file as HTML and the screen as a table
+            self.PrintOutputsHTML(case_data_inputs, case_data_results, self.html_output_file.value)
+        else:
+            # copy the output file to the screen
+            with open(outputfile, encoding='UTF-8') as f:
+                content = f.readlines()  # store all output in one long list
+
+                # Now write each line to the screen
+                for line in content:
+                    sys.stdout.write(line)
+
+    def PrintOutputsHTML(self, inputs, outputs, output_filename: str = 'HIP.html'):
+        """
+        PrintOutputs writes the standard outputs to the output file as HTML. The inputs and outputs are already prepared
+        by the calling function so we just pass them in and use them in writing the HTML. They are dictionaries that
+        contain the already formatted information for output.
+        args:
+            inputs: dict of inputs
+            outputs: dict of outputs
+            output_filename: name of the output file
+        """
+        self.logger.info(f'Init {__class__.__name__!s}: {__name__}')
+
+        try:
+            inputs_table = Table(title="***SUMMARY OF INPUTS***")
+            inputs_table.add_column('Parameter Name', no_wrap=True)
+            inputs_table.add_column('Value', no_wrap=True, justify="center")
+            inputs_table.add_column('Units', no_wrap=True)
+            outputs_table = Table(title="***SUMMARY OF RESULTS***")
+            outputs_table.add_column('Result Name', no_wrap=True)
+            outputs_table.add_column('Value', no_wrap=True, justify="center")
+            outputs_table.add_column('Units', no_wrap=True)
+
+            for key, value in inputs['SUMMARY OF INPUTS'].items():
+                name: str = key
+                val1 = value.strip().split(' ')
+                val = val1[0]
+                unit = ''
+                if len(val1) > 1:
+                    unit: str = str(val1[1])
+                    if '**2' in unit:
+                        unit = unit.replace('**2', '\u00b2')
+                    if '**3' in unit:
+                        unit = unit.replace('**3', '\u00b3')
+                    if 'deg' in unit:
+                        unit = unit.replace('deg', '\u00b0')
```

Comment:
the conditional checks here aren't necessary, you can rewrite as

```
unit = (unit
        .replace('**2', '\u00b2')
        .replace('**3', '\u00b3')
        .replace('deg', '\u00b0'))
```

or equivalent